File: benchmarks/pandas/bench_multi_index_fromarrays.py
"""Benchmark: MultiIndex.from_arrays (pandas equivalent)."""
import json
import time
import pandas as pd

SIZE = 5_000
WARMUP = 3
ITERATIONS = 20

arr1 = [f"a{i % 50}" for i in range(SIZE)]
arr2 = [i % 100 for i in range(SIZE)]

for _ in range(WARMUP):
    pd.MultiIndex.from_arrays([arr1, arr2])

t0 = time.perf_counter()
for _ in range(ITERATIONS):
    pd.MultiIndex.from_arrays([arr1, arr2])
total = (time.perf_counter() - t0) * 1000

print(json.dumps({"function": "multi_index_fromarrays", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_multi_index_fromarrays.ts
/**
 * Benchmark: MultiIndex.fromArrays — build from separate level arrays.
 */
import { MultiIndex } from "../../src/index.js";

const SIZE = 5_000;
const WARMUP = 3;
const ITERATIONS = 20;

const arr1 = Array.from({ length: SIZE }, (_, i) => `a${i % 50}`);
const arr2 = Array.from({ length: SIZE }, (_, i) => i % 100);

for (let i = 0; i < WARMUP; i++) {
  MultiIndex.fromArrays([arr1, arr2]);
}

const t0 = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  MultiIndex.fromArrays([arr1, arr2]);
}
const total = performance.now() - t0;

console.log(
  JSON.stringify({ function: "multi_index_fromarrays", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }),
);
